framework/Models/PostProcessors/validationAlgorithms/PPDSS.py

```diff
@@ -0,0 +1,389 @@
+                                # currently allow raw_values
+    #self.pivotParameter        = None  # list of pivot parameters
+    #self.pivotValues           = []
+    self.pivotParameterFeature = None
+    self.pivotValuesFeature    = []
+    self.pivotParameterTarget  = None
+    self.pivotValuesTarget     = []
+    self.scaleType             = None
+    #self.processTimeRecord     = []
+    #self.scaleType             = ['DataSynthesis','2_2_Affine','Dilation','beta_strain','omega_strain','identity']
+    # assembler objects to be requested
+    self.scaleRatioBeta        = []
+    self.scaleRatioOmega       = []
+    #self.addAssemblerObject('Metric', InputData.Quantity.one_to_infinity)
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the parsed paramInput for this class.
+      @ In, paramInput, ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    super()._handleInput(paramInput)
+    for child in paramInput.subparts:
+      if child.getName() == 'Metric':
+        if 'type' not in child.parameterValues.keys() or 'class' not in child.parameterValues.keys():
+          self.raiseAnError(IOError, 'Tag Metric must have attributes "class" and "type"')
+      elif child.getName() == 'Features':
+        self.features = child.value
+        #self.featuresType = child.parameterValues['type']
+      elif child.getName() == 'Targets':
+        self.targets = child.value
+        #self.TargetsType = child.parameterValues['type']
+      elif child.getName() == 'multiOutput':
+        self.multiOutput = child.value
+      elif child.getName() == 'pivotParameterFeature':
+        self.pivotParameterFeature = child.value
+      elif child.getName() == 'pivotParameterTarget':
+        self.pivotParameterTarget = child.value
+      elif child.getName() == 'scale':
+        self.scaleType = child.value
+      elif child.getName() == 'scaleBeta':
+        self.scaleRatioBeta = child.value
+      elif child.getName() == 'scaleOmega':
+        self.scaleRatioOmega = child.value
+      else:
+        self.raiseAnError(IOError, "Unknown xml node ", child.getName(), " is provided for metric system")
+    #if not self.features:
+    #  self.raiseAnError(IOError, "XML node 'Features' is required but not provided")
+    #elif len(self.features) != len(self.targets):
+    #  self.raiseAnError(IOError, 'The number of variables found in XML node "Features" is not equal the number of variables found in XML node "Targets"')
+
+  def run(self, inputIn):
+    """
+      This method executes the postprocessor action. In this case it loads the
+      results to specified dataObject
+      @ In, inputIn, list, dictionary of data to process
+      @ Out, outputDict, dict, dictionary containing the post-processed results
+    """
+    # assert
+    assert(isinstance(inputIn["Data"], list))
+    assert(isinstance(inputIn["Data"][0][-1], xr.Dataset) and isinstance(inputI["Data"][1][-1], xr.Dataset))
```

Comment:
There is a missing `n` in `inpoutI['Data'][1][-1]`

Reply:
I tried to push it my self but it gave me:
`remote: Password authentication is temporarily disabled as part of a brownout. Please use a personal access token instead.
remote: Please see https://github.blog/2020-07-30-token-authentication-requirements-for-api-and-git-operations/ for more information.
fatal: unable to access 'https://github.com/idaholab/raven.git/': The requested URL returned error: 403`